### PR TITLE
feat: Added `file-annotations-upload` parameter when submitting a manifest

### PR DESCRIPTION
**Pull request by linglp**
## Context
related to: https://sagebionetworks.jira.com/browse/FDS-1785. I added parameter `file-annotations-upload` and provided the option of adding/not adding annotations when submitting a file-based manifest. 

## Monitor the speed
I compared the speed of adding/not adding annotations to entities using schematic profiler. When submitting the same manifest using the example data model, the speed is not always consistent when running the API locally. But on average after running 10 times, for **file-based option** with annotation set to true, the speed is ~15.9s for a file-based manifest with 4 rows and ~14.3s when annotation is set to false with the same manifest. For **table_and_file** option, with annotation set to true, the speed is ~48s while if annotation is set to False, the speed is ~47s. 

## Manual Test
* test without adding the file annotation parameter
`schematic model -c config.yml submit -mrt file_only -d syn55139560 -vc BulkRNA-seqAssay -mp /Users/lpeng/Downloads/Example-bulkrnaseq.csv`

* test with file annotation parameter being set to False with option file_only
`schematic model -c config.yml submit -mrt file_only -d syn55139560 -vc BulkRNA-seqAssay -mp /Users/lpeng/Downloads/Example-bulkrnaseq.csv --no-fa`

* test with file annotation parameter being set to False with option table_and_file
`schematic model -c config.yml submit -mrt table_and_file -d syn55139560 -vc BulkRNA-seqAssay -mp /Users/lpeng/Downloads/Example-bulkrnaseq.csv --no-fa`

* test with file annotation parameter being set to False with option file_and_entities
`schematic model -c config.yml submit -mrt file_and_entities -d syn55139560 -vc BulkRNA-seqAssay -mp /Users/lpeng/Downloads/Example-bulkrnaseq.csv --no-fa`

## Added tests
* Added a test to test manifest submission in CLI under `test_cli.py`
* Added the unit tests for the following functions: 
      * `upload_manifest_combo`
      * `upload_manifest_as_table`
      * `upload_manifest_file`
      * `upload_manifest_as_csv`
      *  `associateMetadataWithFiles`
      * `submit_metadata_manifest`

## Problems noticed
* https://sagebionetworks.jira.com/browse/FDS-1934

### CODE REVIEW

File: schematic/store/synapse.py [hunk 1]
```diff
@@ -1706,6 +1704,7 @@ def upload_manifest_as_table(
         table_manipulation: str,
         table_column_names: str,
         annotation_keys: str,
+        file_annotations_upload: Optional[bool] = True,
```
Comment (andrewelamb) [inline schematic/store/synapse.py:1707]: Does this need to be optional? It seems that True/False is all thats needed here

File: schematic/store/synapse.py [hunk 2]
```diff
@@ -1725,6 +1724,7 @@ def upload_manifest_as_table(
             annotation_keys: (str) display_label/class_label (default), Sets labeling syle for annotation keys. class_label will format the display
                 name as upper camelcase, and strip blacklisted characters, display_label will strip blacklisted characters including spaces, to retain
                 display label formatting while ensuring the label is formatted properly for Synapse annotations.
+            file_annotations_upload (bool): Default to True. If false, do not add annotations to files.
```
Comment (andrewelamb) [inline schematic/store/synapse.py:1727]: Same as above

File: schematic/store/synapse.py [hunk 3]
```diff
@@ -1793,6 +1794,7 @@ def upload_manifest_as_csv(
         hideBlanks,
         component_name,
         annotation_keys: str,
+        file_annotations_upload: Optional[bool] = True,
```
Comment (andrewelamb) [inline schematic/store/synapse.py:1797]: Same as above

File: schematic/store/synapse.py [hunk 4]
```diff
@@ -1851,6 +1855,7 @@ def upload_manifest_combo(
         table_manipulation,
         table_column_names: str,
         annotation_keys: str,
+        file_annotations_upload: Optional[bool] = True,
```
Comment (andrewelamb) [inline schematic/store/synapse.py:1858]: Same as above

File: schematic/store/synapse.py [hunk 5]
```diff
@@ -1934,6 +1941,7 @@ def associateMetadataWithFiles(
         table_manipulation: str = "replace",
         table_column_names: str = "class_label",
         annotation_keys: str = "class_label",
+        file_annotations_upload: Optional[bool] = True,
```
Comment (andrewelamb) [inline schematic/store/synapse.py:1944]: Same as above

File: schematic_api/api/routes.py [hunk 6]
```diff
@@ -392,6 +393,7 @@ def submit_manifest_route(
     project_scope=None,
     table_column_names=None,
     annotation_keys=None,
+    file_annotations_upload:Optional[bool]=True,
```
Comment (andrewelamb) [inline schematic_api/api/routes.py:396]: Same as above

File: schematic/store/synapse.py (hunk 1 above)
Comment (linglp) [reply schematic/store/synapse.py:1707]: I see what you meant. Yes, i could update this to just `file_annotations_upload: bool = True,`

**Review by andrewelamb — APPROVED** [review]

File: schematic/models/commands.py [hunk 7]
```diff
@@ -96,6 +96,13 @@ def model(ctx, config):  # use as `schematic model ...`
     is_flag=True,
     help=query_dict(model_commands, ("model", "validate", "restrict_rules")),
 )
+@click.option(
+    "--file_annotations_upload/--no-file_annotations_upload",
+    "-fa/--no-fa",
```
Comment (mialy-defelice) [inline schematic/models/commands.py:101]: `--no-fa` should be `-no-fa` to follow conventions :)

**Review by mialy-defelice — CHANGES_REQUESTED** [review]
Tested everything, and it works as expected. Just one small change.

File: schematic/models/commands.py (hunk 7 above)
Comment (linglp) [reply schematic/models/commands.py]: Thans @mialy-defelice. I just updated that.

**Review by mialy-defelice — APPROVED** [review]

**Pull request merged**